src/main/java/org/spongepowered/common/data/builder/tileentity/SpongeCommandBlockBuilder.java

```diff
@@ -22,42 +22,44 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.common.data.builder.block.tileentity;
+package org.spongepowered.common.data.builder.tileentity;
 
 import net.minecraft.tileentity.CommandBlockLogic;
 import net.minecraft.tileentity.CommandBlockTileEntity;
 import net.minecraft.tileentity.TileEntity;
-import org.spongepowered.api.block.tileentity.CommandBlock;
+import org.spongepowered.api.block.entity.CommandBlock;
 import org.spongepowered.api.data.persistence.DataView;
 import org.spongepowered.api.data.persistence.InvalidDataException;
+import org.spongepowered.common.accessor.tileentity.CommandBlockLogicAccessor;
 import org.spongepowered.common.text.SpongeTexts;
 import org.spongepowered.common.util.Constants;
 
 import java.util.Optional;
 
-public class SpongeCommandBlockBuilder extends AbstractTileBuilder<CommandBlock> {
+public class SpongeCommandBlockBuilder extends AbstractTileEntityBuilder<CommandBlock> {
 
-    public SpongeCommandBlockBuilder() {
+    protected SpongeCommandBlockBuilder() {
         super(CommandBlock.class, 1);
     }
 
     @Override
-    protected Optional<CommandBlock> buildContent(final DataView container) throws InvalidDataException {
+    protected Optional<CommandBlock> buildContent(DataView container) throws InvalidDataException {
         return super.buildContent(container).flatMap(commandBlock -> {
             if (!container.contains(
-                Constants.TileEntity.CommandBlock.STORED_COMMAND, Constants.TileEntity.CommandBlock.SUCCESS_COUNT, Constants.TileEntity.CommandBlock.DOES_TRACK_OUTPUT)) {
+                    Constants.TileEntity.CommandBlock.STORED_COMMAND, Constants.TileEntity.CommandBlock.SUCCESS_COUNT, Constants.TileEntity.CommandBlock.DOES_TRACK_OUTPUT)) {
                 ((TileEntity) commandBlock).remove();
                 return Optional.empty();
```

Comment:
I almost want these to be funneled through an abstract method so they're not being checked here but rather by the super, so that we don't have unnecessary tile entity creations that end up getting cancelled. Yes, I know the old code did it, but doesn't mean we should continue.

Reply:
Like a sort of `validate` method?